server/clip_server/executors/clip_torch.py

```diff
@@ -108,57 +119,92 @@ async def rank(self, docs: 'DocumentArray', parameters: Dict, **kwargs):
         set_rank(docs)
 
     @requests
-    async def encode(self, docs: 'DocumentArray', parameters: Dict = {}, **kwargs):
-        access_paths = parameters.get('access_paths', self._access_paths)
-        if 'traversal_paths' in parameters:
-            warnings.warn(
-                f'`traversal_paths` is deprecated. Use `access_paths` instead.'
-            )
-            access_paths = parameters['traversal_paths']
-        _drop_image_content = parameters.get('drop_image_content', False)
-
-        _img_da = DocumentArray()
-        _txt_da = DocumentArray()
-        for d in docs[access_paths]:
-            split_img_txt_da(d, _img_da, _txt_da)
-
-        with torch.inference_mode():
-            # for image
-            if _img_da:
-                for minibatch, batch_data in _img_da.map_batch(
-                    partial(
-                        self._preproc_images, drop_image_content=_drop_image_content
-                    ),
-                    batch_size=self._minibatch_size,
-                    pool=self._pool,
-                ):
-                    with self.monitor(
-                        name='encode_images_seconds',
-                        documentation='images encode time in seconds',
-                    ):
-                        minibatch.embeddings = (
-                            self._model.encode_image(**batch_data)
-                            .cpu()
-                            .numpy()
-                            .astype(np.float32)
-                        )
-
-            # for text
-            if _txt_da:
-                for minibatch, batch_data in _txt_da.map_batch(
-                    self._preproc_texts,
-                    batch_size=self._minibatch_size,
-                    pool=self._pool,
-                ):
-                    with self.monitor(
-                        name='encode_texts_seconds',
-                        documentation='texts encode time in seconds',
-                    ):
-                        minibatch.embeddings = (
-                            self._model.encode_text(**batch_data)
-                            .cpu()
-                            .numpy()
-                            .astype(np.float32)
-                        )
+    async def encode(
+        self,
+        docs: 'DocumentArray',
+        tracing_context=None,
```

Comment:
I just wonder how to pass a request specific`tracing_context`  from a client? Is there any doc in jina core?

Reply:
If you're using Flow then you don't need to worry about it. We added [Instrumentation](https://docs.jina.ai/fundamentals/executor/instrumenting-executor/#instrumentation) sub sections for Executor and Flow along with a dedicated [OpenTelemetry Support](https://docs.jina.ai/cloud-nativeness/opentelemetry/) section. 

We haven't documented the `tracing_context` at the Executor level but the [implementation](https://opentelemetry.io/docs/instrumentation/python/cookbook/#manually-setting-span-context) is also not complicated.

Reply:
I see.